File: jvm/src/main/java/ch/neukom/advent2022/day6/Part1.java
package ch.neukom.advent2022.day6;

import java.io.IOException;

import ch.neukom.advent2022.util.InputResourceReader;

import static java.util.stream.Collectors.*;

public class Part1 {
    public static final int START_MARKER_LENGTH = 4;

    public static void main(String[] args) throws IOException {
        try (InputResourceReader reader = new InputResourceReader(Part1.class)) {
            run(reader);
        }
    }

    private static void run(InputResourceReader reader) {
        String input = reader.readInput().collect(joining("\n"));
        int start = Util.findStart(input, START_MARKER_LENGTH);
        System.out.println("The $START_MARKER_LENGTH-character marker starts at $start");
    }
}

File: jvm/src/main/kotlin/ch/neukom/advent2022/day1/Part1.kt
package ch.neukom.advent2022.day1

import ch.neukom.advent2022.util.InputResourceReader
import java.util.stream.Collectors

object Part1

fun main() {
    InputResourceReader(Part1.javaClass).use(::run)
}

private fun run(reader: InputResourceReader) {
    val input = reader.readInput().collect(Collectors.joining("\n"))
    val highestAmountOfCalories = input.splitToSequence("\n\n")
        .map { parseNumbers(it) }
        .map { it.sum() }
        .max()
    println("The highest amount of calories carried is $highestAmountOfCalories")
}

private fun parseNumbers(it: String) = it.splitToSequence("\n").map { it.toLong() }
